vendor/github.com/containers/common/libimage/manifests/manifests.go

```diff
@@ -239,6 +241,16 @@ func (l *list) Push(ctx context.Context, dest types.ImageReference, options Push
 			return nil, "", err
 		}
 	}
+
+	compressionVariants := []cp.CopyOptionCompressionVariant{}
+	//options.EnsureCompressionVariantExists = []CopyOptionCompressionVariant{CopyOptionCompressionVariant{Algorithm: &compression.Zstd, Level:20}}
+	for _, compressionVariant := range options.ReplicateWithCompression {
+		if compressionVariant == "zstd" {
+			variant := cp.CopyOptionCompressionVariant{Algorithm: &compression.Zstd, Level:20}
```

Comment:
Why does this have an opinion on the compression level?

Reply:
The code in this PR is just experimental, i was using this code to test `c/image` but eventually I think most of the things in this PR can be changed and we can have a way to provide `level` at UX level.